gems/sorbet-runtime/lib/types/types/typed_enumerable.rb

```diff
@@ -50,7 +50,11 @@ def valid?(obj)
         # Enumerators can be unbounded: see `[:foo, :bar].cycle`
         return true
       when Range
-        @type.valid?(obj.first) && @type.valid?(obj.last)
+        if obj.end.nil?
+          @type.valid?(obj.begin)
+        else
+          @type.valid?(obj.begin) && @type.valid?(obj.end)
+        end
```

Comment:
Can't we validate the `begin` and `end` separately?
```suggestion
        (obj.begin.nil? || @type.valid?(obj.begin))
          && (obj.end.nil? || @type.valid?(obj.end))
```
This way, any open ends would automatically be valid and if both ends are open, then it would always pass.